lib/actions/merge.js

```diff
@@ -0,0 +1,52 @@
+const { Action } = require('./action')
+
+const checkIfMerged = async (context, prNumber) => {
+  let status = true
+
+  // return can be 204 or 404 only
+  try {
+    await context.github.pulls.checkIfMerged(
+      context.repo({ pull_number: prNumber })
+    )
+  } catch (err) {
+    if (err.status === 404) {
+      status = false
+    } else {
+      throw err
+    }
+  }
+
+  return status
+}
+
+class Comment extends Action {
+  constructor () {
+    super('merge')
+    this.supportedEvents = [
+      'pull_request.*'
+    ]
+  }
+
+  // there is nothing to do
+  async beforeValidate () {}
+
+  async afterValidate (context, settings, results) {
+    const prNumber = this.getPayload(context).number
+    const isMerged = await checkIfMerged(context, prNumber)
+
+    if (!isMerged) {
+      let mergeMethod = settings.merge_method ? settings.merge_method : 'merge'
+      try {
+        await context.github.pulls.merge(context.repo({ pull_number: prNumber, merge_method: mergeMethod }))
+      } catch (err) {
+        // skip on known errors
+        // 405 === Method not allowed , 409 === Conflict
+        if (!(err.status === 405 || err.status === 409)) {
+          throw err
```

Comment:
Can't remember the behavior now, but what is the user experience when this happens? Should we execute a comment to let the user know about failures?

Reply:
We currently have no feature to give feedback when any `action` throw error. right now. even in validators, we only give feedback when it's a `pull_request` by creating a check.

Reply:
I highly recommend putting more thought around a user feedback mechanism now that we are starting to support advanced actions. 

IMO For actions that does not support a feedback mechanism i.e. `check` we use comments in within the PR/Issue.

Reply:
yes, i've been thinking about it. I think it should go into a separate PR that handles all `action` errors. since that scope is bigger than just one action. I'll fix PR this so that it will throw proper error

Reply:
Agreed.

Reply:
fixed